src/documents/Contexts.md

```diff
@@ -61,15 +52,106 @@ at a particular component location in the tree are the same, as described in the
 ## Where context names are looked for ##
 
 Context names listed in a configuration are searched for at instantiation time, sequentially,
-in three kinds of "scopes" or "environments".
+in two kinds of "scopes" or "environments".
 
 * Most immediately, context names are searched for in the tree of currently instantiating
 components, searching __upwards__ through the tree from the parent of the component where the context
 name is held.
-* Following this, a search is made in the *dynamic environment*, which is notionally a separate tree
-of components bound to the currently executing stack frame (this facility is deprecated and will be
-removed in a future version of the framework)
-* Finally, context names are sought in the *static environment*, which is a tree of components
-attached globally to the JS VM (or context within it) as a whole.
+* Following this, a search is made in the *resolver root*, which is notionally a separate tree
+of components which holds all components which has been instantiated with the grade `fluid.resolveRoot` anywhere in the
+current component tree.
 
 ![Diagram showing coloured squared representing components in a component tree](images/IoC-scope.svg)
+
+## Global components: `fluid.resolveRoot` and `fluid.resolveRootSingle`
+
+All Infusion components are stored in a single-rooted **_component tree_** - this stores every component which
+has been instantiated in a particular **_Infusion context_**. This context is global to the JavaScript context 
+which has loaded Infusion - for example, a frame within a browser, or a node.js process. Even when you 
```

Comment:
This opening statement is pretty helpful in understanding Contexts. I think it should be placed much closer to the opening paragraph of the page, where "component tree" is first mentioned.